connector/loganomalyconnector/connector.go

```diff
@@ -0,0 +1,147 @@
+// Copyright observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package loganomalyconnector
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+)
+
+type logBatch struct {
+	logs    plog.Logs
+	ctx     context.Context
+	errChan chan error
+}
+
+// Detector implements the log anomaly detection connector.
+// It maintains a rolling window of log rate samples and uses statistical analysis
+// to detect anomalies in log throughput. The detector processes incoming logs,
+// calculates rates, and generates alerts when anomalous patterns are detected.
+type Detector struct {
+	ctx    context.Context
+	cancel context.CancelFunc
+	logger *zap.Logger
+
+	stateLock sync.Mutex
+	config    *Config
+
+	counts            []int64
+	lastWindowEndTime time.Time
+
+	logChan      chan logBatch
+	nextConsumer consumer.Logs
+}
+
+func newDetector(config *Config, logger *zap.Logger, nextConsumer consumer.Logs) *Detector {
+	ctx, cancel := context.WithCancel(context.Background())
+	numWindows := int(config.MaxWindowAge.Minutes())
+
+	return &Detector{
+		ctx:               ctx,
+		cancel:            cancel,
+		logger:            logger,
+		config:            config,
+		stateLock:         sync.Mutex{},
+		nextConsumer:      nextConsumer,
+		logChan:           make(chan logBatch, 1000),
+		counts:            make([]int64, numWindows),
+		lastWindowEndTime: time.Now().Truncate(time.Minute),
+	}
+}
+
+// Start begins the anomaly detection process, sampling at intervals specified in the config.
+// It launches a background goroutine that periodically checks for and updates anomalies.
+func (d *Detector) Start(_ context.Context, _ component.Host) error {
+
+	ticker := time.NewTicker(d.config.SampleInterval)
+
+	go func() {
+		for {
+			select {
+			case <-d.ctx.Done():
+				return
+			case batch := <-d.logChan:
+				d.processLogBatch(batch)
+			case <-ticker.C:
+				d.checkForAnomaly() // check this
+			}
+		}
+	}()
+
+	return nil
+}
+
+func (d *Detector) processLogBatch(batch logBatch) {
+	now := time.Now()
+
+	// Update windows that have elapsed since last update
+	for now.Sub(d.lastWindowEndTime).Minutes() >= 1 {
+		// Shift windows and add empty window
+		copy(d.counts, d.counts[1:])
```

Comment:
Since we're reading the counts in a separate goroutine, we need to be threadsafe. Maybe it best to use accumulate log counts onto an atomic int, and have the ticker trigger all updates to the count slice.